### DISCUSSION TITLE
Prisma client not working when deployed over AWS using serverless

### DISCUSSION THREAD

**Discussion by amitleuva1987** [prisma/prisma#14238] [Q&A] [answered]
### Bug description

Below is the URL to my git repository, and read me contains all steps to reproduce the issue.

[git repository](https://github.com/amitleuva1987/full-stack-nextjs-with-prisma)

when I deploy this repository over aws using serverless, prisma client does not work. I find the related solutions mentioned in the serverless documents and other repos which did not help. Below is one of the solution which looked promising but it also did not work.

[millsp prisma serverless next js - repository url](https://github.com/millsp/prisma-serverless-nextjs)

I have also posted an issue with above repository. I have also find an open issue by few other users which needs help.

[git issue](https://github.com/prisma/prisma/issues/12702)

Thanks 

### How to reproduce

Below is the URL to my git repository, and read me contains all steps to reproduce the issue.

[git repository](https://github.com/amitleuva1987/full-stack-nextjs-with-prisma)

Open the next.config.js file and fill the environment variables.

```
NEXTAUTH_URL:'******',
GIT_ID:"*******",
GIT_SECRET:"*******",
EMAIL_SERVER:"*********",
EMAIL_FROM:"********",
DATABASE_URL:"********"
```
then run below commands

```
npm install
npx prisma generate
npm run build
serverless
```

### Expected behavior

prisma client should work without 500 internal server error. .next_serverless folder should have prisma client in order for prisma to work in the app

### Prisma information

DATABASE_URL:"********"

### Environment & setup

Open the next.config.js file and fill the environment variables.

```
NEXTAUTH_URL:'******',
GIT_ID:"*******",
GIT_SECRET:"*******",
EMAIL_SERVER:"*********",
EMAIL_FROM:"********",
DATABASE_URL:"********"

### Prisma Version

```
"prisma": "^3.15.2"
"@prisma/client": "^3.15.2",
"next": "12.1.6",
```

**Comment by janpio** [comment]
What does "does not work" mean in your case?

**Reply by amitleuva1987** [reply]
In my case means, I use "windows 10" , "prisma": "^3.15.2", "@prisma/client": "^3.15.2", "next": "12.1.6". It does not work with these stacks. There is no error while I deploy over AWS but when i visit the page, it says "500 internal server error".  I tried [milips solution](https://github.com/millsp/prisma-serverless-nextjs). But, the postBuildCommands in serverless.yml file throws error when I execute serverless. below is serverless.yml code

```
yourProjectName:
  component: "@sls-next/serverless-component@1.19.0"
  inputs:
    minifyHandlers: true
    build:
      postBuildCommands:
        - PDIR=node_modules/.prisma/client/;
          LDIR=.serverless_nextjs/api-lambda/;
          if [ "$(ls -A $LDIR)" ]; then
            mkdir -p $LDIR$PDIR;
            cp "$PDIR"query-engine-rhel-* $LDIR$PDIR;
            cp "$PDIR"schema.prisma $LDIR$PDIR;
          fi;
        - PDIR=node_modules/.prisma/client/;
          LDIR=.serverless_nextjs/default-lambda/;
            if [ "$(ls -A $LDIR)" ]; then
              mkdir -p $LDIR$PDIR;
              cp "$PDIR"query-engine-rhel-* $LDIR$PDIR;
              cp "$PDIR"schema.prisma $LDIR$PDIR;
          fi;
```

above code actually copies "node_modules/.prisma/client" folder into "serverless_nextjs/api-lambda" and "serverless_nextjs/default-lambda". When, I execute serverless,  it says "'PDIR' is not recognized as an internal or external command, operable program or batch file. ". I have also opened an [issue ](https://github.com/millsp/prisma-serverless-nextjs/issues/2) regarding this.

Please let me know if I am using any outdated solution since millips proposed solution is more than a year old.

**Reply by janpio** [reply]
That code does not work on Windows.

**Comment by amitleuva1987** [comment]
Is there a way to make it run on windows?

**Comment by amitleuva1987** [comment, marked as answer]
This issue has been solved. this is the [solution url](https://github.com/serverless-nextjs/serverless-next.js/issues/2459#event-6999739076)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
